plugins/net_plugin/net_plugin.cpp

```diff
@@ -1965,6 +1965,9 @@ namespace eosio {
 
          if( set_state_to_head_catchup ) {
             set_state( head_catchup );
+            if( blk_num == sync_last_requested_num ) {
+                request_next_chunk( std::move( g_sync ));
```

Comment:
I have not looked at the logic of this, but you do need to lock `g_sync` before the call. `request_next_chunk` assumes `g_sync` is locked. It was unlocked on line 1945 above.

Reply:
`g_sync` needs to be locked before accessing `sync_last_requested_num` as that var is protected by `sync_mtx`.

Reply:
killing this pull request and creating a new one